lsp-mode.el

```diff
@@ -6303,15 +6303,81 @@ representation to point representation."
                   (< c1 c2))
                  (c1 t)))))
 
-(defun lsp--imenu-create-index ()
-  "Create imenu index from document symbols."
-  (let* ((filtered-symbols (lsp--imenu-filter-symbols (lsp--get-document-symbols)))
-         (lsp--line-col-to-point-hash-table (-> filtered-symbols
+(defun lsp-imenu-create-uncategorized-index (symbols)
+  "Create imenu index from document SYMBOLS."
+  (let* ((lsp--line-col-to-point-hash-table (-> symbols
                                                 lsp--collect-lines-and-cols
                                                 lsp--convert-line-col-to-points-batch)))
-    (if (lsp--imenu-hierarchical-p filtered-symbols)
-        (lsp--imenu-create-hierarchical-index filtered-symbols)
-      (lsp--imenu-create-non-hierarchical-index filtered-symbols))))
+    (if (lsp--imenu-hierarchical-p symbols)
+        (lsp--imenu-create-hierarchical-index symbols)
+      (lsp--imenu-create-non-hierarchical-index symbols))))
+
+(defcustom lsp-imenu-symbol-categories
```

Comment:
Should I `lsp--symbol-kind` here? I wanted to make this at least somewhat customizable (I had in mind that a user could  replace the strings with icons, for example).

Reply:
Also, should we make `lsp--symbol-kind` defcustom?

Reply:
Seems both have the same info and it will be better to use the same variable. It can be changed to defcustom.

Reply:
I made that one defcustom. However, I believe having a separate alist is the proper way to do it, as it allows pluralized categories (as is imenu convention, apparently).